### PR TITLE
[LM-10472] Added Notification Feed Screen and its Styles and Customizations

### PR DESCRIPTION
**Pull request by sidcr7-likeminds**
## Description

**Features**

- Added Notification Feed Screen
- Added styles related to Notification Feed Screen
- Added customizations related to Notification Feed Screen

**Bugs**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Hotfix (fix for breaking changes)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Other, please describe:

### Author Checklist

- [x] Assignee added
- [x] Reviewer added
- [x] Description added
- [ ] Unit test cases added

### Reviewer Checklist

- [x] PR title starts with Jira ticket number prefix
- [ ] If hotfix, Add Hotfix as prefix in PR title
- [x] Label added
- [x] Commit messages follows standard format (added, changed, removed, code commented and chore)
- [x] Description about the Feature/ Bugs added

### CODE REVIEW

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/adapter/LMFeedActivityFeedAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+        val itemNotificationFeedViewDataBinder = LMFeedItemActivityFeedViewDataBinder(listener)
+        viewDataBinders.add(itemNotificationFeedViewDataBinder)
+
+        return viewDataBinders
+    }
+}
+
+interface LMFeedActivityFeedAdapterListener {
+    fun onActivityFeedItemClicked(position: Int, activityViewData: LMFeedActivityViewData)
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/adapter/LMFeedActivityFeedAdapter.kt:26]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = LmFeedFragmentActivityFeedBinding.inflate(layoutInflater)
+
+        binding.apply {
+            customizeActivityFeedHeaderView(headerViewActivityFeed)
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt:37]: function to customize list view is missing

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+                )
+            }
+
+            return root
+        }
+    }
+
+    protected open fun customizeActivityFeedHeaderView(headerViewActivityFeed: LMFeedHeaderView) {
+        headerViewActivityFeed.apply {
+            setStyle(LMFeedStyleTransformer.activityFeedFragmentViewStyle.headerViewStyle)
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt:57]: check lint

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    private fun checkForNoActivity(feed: List<LMFeedBaseViewType>) {
+        if (feed.isNotEmpty()) {
+            binding.apply {
+                //todo:
+//                layoutNoNotification.root.hide()
+                rvActivityFeed.show()
+            }
+        } else {
+            binding.apply {
+//                layoutNoNotification.root.show()
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt:176]: why this is commented?

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedListView.kt
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedListView.kt:1]: add comments to public functions

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+                        color
+                    )
+                )
+            }
+
+            return root
+        }
+    }
+
+    protected open fun customizeActivityFeedHeaderView(headerViewActivityFeed: LMFeedHeaderView) {
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt:55]: add comments

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/universalfeed/view/LMFeedUniversalFeedFragment.kt [hunk 6]
```diff
@@ -468,6 +463,8 @@ open class LMFeedUniversalFeedFragment : Fragment(), LMFeedUniversalFeedAdapterL
     }
 
     protected open fun onSearchIconClick() {
+        //todo: change this
```
Comment (ishaan1607) [inline likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/universalfeed/view/LMFeedUniversalFeedFragment.kt:466]: why it is on search icon clicked

**Review by ishaan1607 — CHANGES_REQUESTED** [review]

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt (hunk 2 above)
Comment (sidcr7-likeminds) [reply likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt:37]: add this in TODOs will add this while doing docs

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt (hunk 3 above)
Comment (sidcr7-likeminds) [reply likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt:57]: it looks good as per Android Studio linting

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt (hunk 4 above)
Comment (sidcr7-likeminds) [reply likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/activityfeed/view/LMFeedActivityFeedFragment.kt:176]: added in further PRs. The no layout view was not ready till this time

File: likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/universalfeed/view/LMFeedUniversalFeedFragment.kt (hunk 6 above)
Comment (sidcr7-likeminds) [reply likeminds-feed-android-core/src/main/java/com/likeminds/feed/android/core/universalfeed/view/LMFeedUniversalFeedFragment.kt:466]: it was for testing only, this is fixed and moved to proper function in further PRs

**Review by ishaan1607 — APPROVED** [review]

**Pull request merged**